CMakeLists.txt

```diff
@@ -35,8 +32,6 @@ set_pennylane_lightning_version(${PROJECT_SOURCE_DIR}/pennylane_lightning/_versi
 message(STATUS "pennylane_lightning version ${VERSION_STRING}")
 set(PROJECT_VERSION ${VERSION_STRING})
 
-set(CMAKE_CXX_STANDARD 17) # At least C++17 is required
```

Comment:
Why not 
```
 set(CMAKE_CXX_STANDARD 20) # At least C++20 is required
```
?

Reply:
@maliasadi As compilers on our target OSs support C++20 (even very partially), the purpose of the current PR is to move to C++20 experimentally. As we serve compiled wheels to all users, I think it is okayish to move C++20. We may roll back this requirement when a fatal problem is discovered.

Reply:
Agreed with Chae-Yeun. It helps simplify many parts of the codebase and remove some utilities we can defer to the standard, so provided those all work, I think we are good.

Reply:
Thanks guys! Sure it's a good move then.